src/app.js

```diff
@@ -347,6 +347,21 @@ app.routeAsync("/groups/:groupName/settings/overview")
   .optionsAsync(optionsGroup)
 ;
 
+app.routeAsync("/groups/:groupName/settings/members")
+  .getAsync(endpoints.groups.listMembers);
+
+app.routeAsync("/groups/:groupName/settings/roles")
+  .getAsync(endpoints.groups.listRoles);
+
+app.routeAsync("/groups/:groupName/settings/roles/:roleName/members")
+  .getAsync(endpoints.groups.listRoleMembers);
+
+app.routeAsync("/groups/:groupName/settings/roles/:roleName/members/:username")
+  .getAsync(endpoints.groups.getRoleMember)
+  .putAsync(endpoints.groups.putRoleMember)
+  .deleteAsync(endpoints.groups.deleteRoleMember)
```

Comment:
I noticed that after every PUT/DELETE, I would get `UnauthorizedError`s from [here](https://github.com/nextstrain/nextstrain.org/blob/ad853a2445d6fd726ada036b73cc05608cf00011/src/endpoints/groups.js#L13) until I restarted the server. I haven't looked deeper, so not sure if it's a problem.

Reply:
Ahhh. This is expected in this case because you're operating on your own user. Changing the role of a user X marks as stale any token X was issued before the change. X must renew/refresh their token to continue making authenticated requests. This happens automatically and transparently in a web context (e.g. if client-side JS was making the API requests) because the server manages the tokens, but here you're managing them. Nextstrain CLI [detects this condition and prompts the user](https://github.com/nextstrain/cli/blob/4797a5fee9631e034be76377e243b7d46d220bfc/nextstrain/cli/remote/nextstrain_dot_org.py#L715-L736) to `nextstrain login --renew` and then retry, but it could also make the renewal automatic and transparent in the future.

A server restart only fixed it by happenstance because in local dev you don't have Redis to persist the "stale before" bits, so an in-process memory store is used. This is one bit of janky test/dev set up that would be nice to improve at some point so parity with production is better. We could totally be using Redis locally in dev more routinely; I only use it now when I want to specifically test the Redis bits. But not that high priority to address it yet.

Reply:
Got it! Thanks for the detailed explanation.